components/python/pygobject-34/pygobject-34-GENFRAG.p5m

```diff
@@ -10,13 +10,16 @@
 #
 
 #
+# Copyright 2021 Gary Mills
 # Copyright 2016 Alexander Pyhalov
 #
 
 license COPYING license="LGPL v2.1"
 
-file path=usr/include/pygobject-3.0/pygobject.h
-file path=usr/lib/$(MACH64)/pkgconfig/pygobject-3.0.pc
+# Manually added since python2 version is built in ../pygobject-3
+depend type=conditional fmri=$(COMPONENT_FMRI)-27 \
+    predicate=runtime/python-27
 
-depend type=require \
-    fmri=library/python/pycairo@1.10.0
+# Manually added since python3 version is also built in ../pygobject-3
+depend type=conditional fmri=$(COMPONENT_FMRI)-35 \
+    predicate=runtime/python-35
```

Comment:
These two dependencies look wrong. The python runtime dependencies belong to the corresponding manifest files. And the dependencies for python runtimes 3.7 and 3.9 seem to miss.

Reply:
They seem correct to me, for a conditional dependency.  The condition is that the package runtime/python-35 be installed.  When I do "pkg contents -m library/python/pygobject-3-35 | grep depend", I get "depend fmri=library/python/pygobject-3 type=require".  I have the package runtime/python-35 installed on my build server.

Reply:
But you are forcing runtime dependencies for python 2.7 and 3.5 in your pygobject-34 packages where you only provide package versions for python 3.7 and 3.9. The require dependencies should be were the packages are delivered. So, pygobjet-3-35 should require python 3.5, pygobject-3-37 should require python 3.7 and so on. You can look in other python packages, eg. setuptools, how it is done in the PYVER manifest:

# force a dependency on the Python $(PYVER) runtime
depend type=require fmri=__TBD pkg.debug.depend.file=python$(PYVER) \
    pkg.debug.depend.path=usr/bin

Reply:
Ah, I see the problem.  The conditional dependencies are correct, but an unrelated dependency is missing.  I compared the pygobject-3-35 and pygobject-3-37 packages, generated from different manifests.  Both pygobject-3-35 and -27 packages are missing an explicit dependency on runtime/python-35 or -27, respectively.  I will correct that omission in the next push.